plugins/hypervisors/xenserver/test/com/cloud/hypervisor/xenserver/resource/wrapper/xenbase/CitrixRequestWrapperTest.java

```diff
@@ -1320,14 +1322,23 @@ public void testOvsDestroyTunnelCommandFailed() {
 
     @Test
     public void testUpdateHostPasswordCommand() {
-        final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123");
+        final ExecutionResult executionResult = Mockito.mock(ExecutionResult.class);
+
+        final UpdateHostPasswordCommand updatePwd = new UpdateHostPasswordCommand("test", "123", "127.0.0.1");
+
+        final StringBuffer buff = new StringBuffer();
+        buff.append(updatePwd.getUsername());
+        buff.append(' ');
+        buff.append(updatePwd.getNewPassword());
+
+        when(citrixResourceBase.executeInVR(updatePwd.getHostIp(), VRScripts.UPDATE_HOST_PASSWD, buff.toString())).thenReturn(executionResult);
```

Comment:
Hi @karuturi 

The tests is not checking that we can SSH into the host itself, but just covering the flow we have with the new wrappers. That's why we have the Mock of ExecutionResult there.

The unit tests is covering the call CitrixRequestWrapper.execute() when a UpdateHostPasswordCommand is given, not the SSH connection itself. The latter should be done separately.

Thanks for the reivew.

Cheers,
Wilder